src/lib/Behat/PageElement/Fields/RichText.php

```diff
@@ -57,4 +58,103 @@ public function verifyValueInItemView(array $values): void
     }
+
+    public function openElementsToolbar()
+    {
+        $this->context->findElement($this->fields['addButton'])->click();
+    }
+
+    public function changeStyle(string $style)
+    {
+        if (!in_array($style, self::ALLOWED_STYLE_OPTIONS)) {
+            throw new Exception(sprintf('Unsupported style: %s', $style));
+        }
+
+        $this->context->findElement($this->fields['styleDropdown'])->click();
+        $this->context->findElement(sprintf('.ae-listbox li %s', $style))->click();
+    }
+
+    public function insertNewLine(): void
+    {
+        $this->getFieldInput();
+        $this->context->getSession()->getDriver()->executeScript(sprintf($this->executeAlloyEditorScript, $this->richtextId, 'enter'));
+    }
+
+    public function insertLine($value, $style = ''): void
+    {
+        $this->getFieldInput();
+        $this->context->getSession()->getDriver()->executeScript(sprintf($this->insertAlloyEditorValueScript, $this->richtextId, $value));
+
+        if ($style !== '') {
+            $this->changeStyle($style);
+            Assert::assertContains(sprintf('%s%s</%s>', $value, '<br>', $style), $this->context->findElement(sprintf('.ez-data-source__richtext %s', $style))->getOuterHtml());
+        }
+    }
+
+    private function getFieldInput(): NodeElement
+    {
+        $fieldInput = $this->context->findElement($this->fields['fieldInput']);
+        Assert::assertNotNull($fieldInput, sprintf('Input for field %s not found.', $this->label));
+        $this->richtextId = $fieldInput->getAttribute('id');
+
+        return $fieldInput;
+    }
+
+    public function clickOnRichTextEditor()
+    {
+        $fieldInput = $this->getFieldInput();
+        $fieldInput->click();
+    }
+
+    public function addUnorderedList(array $listElements)
+    {
+        $this->getFieldInput();
+        $this->openElementsToolbar();
+        $this->context->findElement($this->fields['unorderedListButton'])->click();
+
+        foreach ($listElements as $listElement) {
+            $this->insertLine($listElement);
+
+            if ($listElement !== end($listElements)) {
+                $this->insertNewLine();
+            }
+        }
+
+        $actualListElements = $this->context->findAllElements($this->fields['unorderedList']);
+        $listElementsText = [];
+        foreach ($actualListElements as $actualListElement) {
+            $listElementsText[] = $actualListElement->getText();
+        }
+
+        Assert::assertEquals($listElements, $listElementsText);
+    }
+
+    public function clickEmbedInlineButton()
+    {
+        $this->context->findElement($this->fields['embedInlineButton'])->click();
+    }
+
+    public function clickEmbedButton()
+    {
+        $this->context->findElement($this->fields['embedButton'])->click();
+    }
+
+    public function containsEmbedInlineItem($itemName)
+    {
+        Assert::assertContains($itemName, $this->context->findElement($this->fields['embedInlineTitle'])->getText());
```

Comment:
This is different from the convention that we're using, that is:
- if the function asserts something then it should be named `verify....` (verifyContainsEmbedInlineItem in this case).
- if the function name is a question (`contains Embed Inline item?`) then it should return a boolean value (`return $itemName === $this->context->findElement($this->fields['embedInlineTitle'])->getText()`) and the assertion should happen here: https://github.com/ezsystems/ezplatform-page-builder/pull/421/files#diff-ab43e5513485e9364a60c01a53e26debR86

I'd go with a boolean here and put the assertion in PB. This comment also applies to `containsEmbedItem`.